shapely/constructive.py

```diff
@@ -902,18 +914,47 @@ def snap(geometry, reference, tolerance, **kwargs):
 
     Examples
     --------
-    >>> from shapely import LineString, Point, Polygon
+    >>> from shapely import snap, distance, LineString, Point, Polygon, MultiPolygon, MultiPoint, box
+
     >>> point = Point(0.5, 2.5)
     >>> target_point = Point(0, 2)
     >>> snap(point, target_point, tolerance=1)
     <POINT (0 2)>
     >>> snap(point, target_point, tolerance=0.49)
     <POINT (0.5 2.5)>
+
     >>> polygon = Polygon([(0, 0), (0, 10), (10, 10), (10, 0), (0, 0)])
     >>> snap(polygon, Point(8, 10), tolerance=5)
     <POLYGON ((0 0, 0 10, 8 10, 10 0, 0 0))>
     >>> snap(polygon, LineString([(8, 10), (8, 0)]), tolerance=5)
     <POLYGON ((0 0, 0 10, 8 10, 8 0, 0 0))>
+
+    You can snap one line to another, for example to clean imprecise coordinates:
+
+    >>> line1 = LineString([(0.1, 0.1), (0.49, 0.51), (1.01, 0.89)])
+    >>> line2 = LineString([(0, 0), (0.5, 0.5), (1.0, 1.0)])
+    >>> snap(line1, line2, 0.25)
+    <LINESTRING (0 0, 0.5 0.5, 1 1)>
+
+    # XYZ geometry:
+    >>> point1 = Point(0.1, 0.1, 0.5)
+    >>> multipoint = MultiPoint([(0, 0, 1), (0, 0, 0)])
+    >>> snap(point1, multipoint, 1)
+    <POINT Z (0 0 1)>
+
+    # empty geometry:
+    >>> snap(line1, LineString([]), 0.25)
+    <LINESTRING (0.1 0.1, 0.49 0.51, 1.01 0.89)>
+
+    >>> snap(line1, None, 0.25) is None
+    True
+
+    # example that shows how snapping only updates one of the coordinates in a Polygon
```

Comment:
```suggestion
    Only one vertex of a polygon is snapped to a target point, even if all vertices are equidistant to it, in order to prevent collapse of the polygon:

    >>> poly = box(0, 0, 1, 1)
    >>> poly
    <POLYGON ((1 0, 1 1, 0 1, 0 0, 1 0))>
    >>> snap(poly, Point(0.5, 0.5), 1)
    <POLYGON ((0.5 0.5, 1 1, 0 1, 0 0, 0.5 0.5))>
```

However, this might be an awkward example to communicate, so it may be OK to leave it out.